File: CS/WindowsApplication/LicensePopup.cs
using System;
using System.Windows.Forms;
using Habanero.Licensing.Validation;

namespace WindowsApplication
{
    public partial class LicensePopup : Form
    {
        public LicensePopup()
        {
            InitializeComponent();

            //load current license information
            LicenseValidationResult valResult = Common.FileValidator.CheckLicense();

            UpdateUI(valResult);
        }



        private void btnActivateTrial_Click(object sender, EventArgs e)
        {
            LicenseValidationResult valResult = Common.FileValidator.ActivateTrial(10);
            Close();
        }

        private void UpdateUI(LicenseValidationResult valResult)
        {
            //we can change the ui based on the existing license here
            if (valResult.State == LicenseState.Invalid)
            {
                btnActivateTrial.Enabled = true;
            }
            else
            {
                btnActivateTrial.Enabled = false;
            }

            //load the rawlicense into a textbox
            txtLicense.Text = valResult.RawLicenseData;
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            //check the new license
            LicenseValidationResult valResult = Common.FileValidator.CheckLicense(txtLicense.Text);
            if (valResult.State == LicenseState.Invalid)
            {
                MessageBox.Show("The license you entered is not valid and will not be saved!", "Invalid License", MessageBoxButtons.OK);
            }
            else
            {
                Common.FileValidator.SaveLicense(txtLicense.Text);
                Close();
            }
        }

        private void btnClear_Click(object sender, EventArgs e)
        {
            Common.FileValidator.SaveLicense("");
            Close();
        }
       
    }
}

File: SampleAppVB/LicensePopup.vb
Imports Habanero.Licensing.Validation

Public Class LicensePopup

    Private Sub btnClear_Click(sender As System.Object, e As System.EventArgs) Handles btnClear.Click
        Common.FileValidator.SaveLicense("")
        Close()
    End Sub

    Private Sub btnActivateTrial_Click(sender As System.Object, e As System.EventArgs) Handles btnActivateTrial.Click
        Dim valResult As LicenseValidationResult = Common.FileValidator.ActivateTrial(10)
        Close()
    End Sub

    Private Sub btnSave_Click(sender As System.Object, e As System.EventArgs) Handles btnSave.Click
        'check the new license
        Dim valResult As LicenseValidationResult = Common.FileValidator.CheckLicense(txtLicense.Text)
        If valResult.State = LicenseState.Invalid Then
            MessageBox.Show("The license you entered is not valid and will not be saved!", "Invalid License", MessageBoxButtons.OK)
        Else
            Common.FileValidator.SaveLicense(txtLicense.Text)
            Close()
        End If
    End Sub

    Private Sub LicensePopup_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        'load current license information
        Dim valResult As LicenseValidationResult = Common.FileValidator.CheckLicense()

        UpdateUI(valResult)
    End Sub

    Private Sub UpdateUI(valResult As LicenseValidationResult)
        'we can change the ui based on the existing license here
        If valResult.State = LicenseState.Invalid Then
            btnActivateTrial.Enabled = True
        Else
            btnActivateTrial.Enabled = False
        End If

        'load the rawlicense into a textbox
        txtLicense.Text = valResult.RawLicenseData
    End Sub
End Class